libraries/chain/abi_serializer.cpp

```diff
@@ -228,13 +237,35 @@ namespace eosio { namespace chain {
          return type;
    }
 
+   // true if std::string can be converted to name
+   bool abi_serializer::is_string_valid_name(std::string_view str) const
+   {
+      if (size_t len = str.size(); len > 13)
```

Comment:
Curious why we need to store the size in a variable and then do the comparison instead of just comparing the value returned from `str.size()`